test/TestSrt.cpp

```diff
@@ -0,0 +1,231 @@
+
+    std::vector<uint8_t> sendBuffer(1000);
+    std::condition_variable serverCondition;
+    std::mutex serverMutex;
+    bool serverGotData = false;
+    SRTSOCKET clientSocket;
+    server.receivedData = [&](std::unique_ptr<std::vector<uint8_t>>& data, SRT_MSGCTRL& msgCtrl,
+                              std::shared_ptr<SRTNet::NetworkConnection>& ctx, SRTSOCKET socket) {
+        EXPECT_EQ(ctx, serverCtx);
+        EXPECT_EQ(*data, sendBuffer);
+        clientSocket = socket;
+        {
+            std::lock_guard<std::mutex> lock(serverMutex);
+            serverGotData = true;
+        }
+        SRT_MSGCTRL msgSendCtrl = srt_msgctrl_default;
+        EXPECT_TRUE(server.sendData(data->data(), data->size(), &msgSendCtrl, socket));
+        serverCondition.notify_one();
+        return true;
+    };
+
+    std::condition_variable clientCondition;
+    std::mutex clientMutex;
+    bool clientGotData = false;
+    client.receivedData = [&](std::unique_ptr<std::vector<uint8_t>>& data, SRT_MSGCTRL& msgCtrl,
+                              std::shared_ptr<SRTNet::NetworkConnection>& ctx, SRTSOCKET socket) {
+        EXPECT_EQ(ctx, clientCtx);
+        EXPECT_EQ(*data, sendBuffer);
+        {
+            std::lock_guard<std::mutex> lock(clientMutex);
+            clientGotData = true;
+        }
+        clientCondition.notify_one();
+        return true;
+    };
+
+    std::fill(sendBuffer.begin(), sendBuffer.end(), 1);
+    SRT_MSGCTRL msgCtrl = srt_msgctrl_default;
+    EXPECT_TRUE(client.sendData(sendBuffer.data(), sendBuffer.size(), &msgCtrl));
+
+    {
+        std::unique_lock<std::mutex> lock(serverMutex);
+        bool successfulWait = serverCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return serverGotData; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for receiving data from client";
+    }
+
+    {
+        std::unique_lock<std::mutex> lock(clientMutex);
+        bool successfulWait = clientCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return clientGotData; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for receiving data from server";
+    }
+
+    SRT_TRACEBSTATS clientStats;
+    client.getStatistics(&clientStats, SRTNetClearStats::no, SRTNetInstant::yes);
+    SRT_TRACEBSTATS serverStats;
+    server.getStatistics(&serverStats, SRTNetClearStats::no, SRTNetInstant::yes, clientSocket);
+    EXPECT_EQ(clientStats.pktSentTotal, 1);
+    EXPECT_EQ(clientStats.pktRecvTotal, 1);
+    EXPECT_EQ(clientStats.pktSentTotal, serverStats.pktRecvTotal);
+    EXPECT_EQ(clientStats.pktRecvTotal, serverStats.pktSentTotal);
+}
+
+// TODO Enable test when STAR-238 is fixed
+TEST(TestSrt, DISABLED_RejectConnection) {
+    SRTNet server;
+    SRTNet client;
+
+    std::condition_variable connectedCondition;
+    std::mutex connectedMutex;
+    bool connected = false;
+
+    server.clientConnected = [&](struct sockaddr& sin, SRTSOCKET newSocket,
```

Comment:
> (The name of this callback should maybe have been validatePendingConnection instead of clientConnected...)

Yes, but "clientConnected" is also a good descriptive name, you do not necessarily have to validate a client.